google/auth/_refresh_worker.py

```diff
@@ -13,39 +13,29 @@
 # limitations under the License.
 
 import logging
-import queue
 import threading
 
 import google.auth.exceptions as e
 
-WORKER_TIMEOUT_SECONDS = 5
-
 _LOGGER = logging.getLogger(__name__)
 
 
-class RefreshWorker:
+class RefreshThreadManager:
     """
-    A worker that will perform a non-blocking refresh of credentials.
+    Organizes exactly one background job that refresh a token.
     """
 
-    MAX_REFRESH_QUEUE_SIZE = 1
-    MAX_ERROR_QUEUE_SIZE = 2
-
     def __init__(self):
-        """Initializes the worker."""
+        """Initializes the manager."""
 
-        self._refresh_queue = queue.Queue(self.MAX_REFRESH_QUEUE_SIZE)
-        # Bound the error queue to avoid infinitely growing the heap.
-        self._error_queue = queue.Queue(self.MAX_ERROR_QUEUE_SIZE)
         self._worker = None
+        self._lock = threading.Lock()
```

Comment:
self._worker_lock = threading.Lock() # protects all access to _worker.

Reply:
Added this comment.